components/list/list-item-link-mixin.js

```diff
@@ -3,6 +3,55 @@ import { css, html } from 'lit';
 import { getUniqueId } from '../../helpers/uniqueId.js';
 import { ListItemMixin } from './list-item-mixin.js';
 
+const interactiveElements = {
+	// 'a' only if an href is present
+	'button': true,
+	'input': true,
+	'select': true,
+	'textarea': true
+};
+
+const interactiveRoles = {
+	'button': true,
+	'checkbox': true,
+	'combobox': true,
+	'link': true,
+	'listbox': true,
+	'menuitem': true,
+	'menuitemcheckbox': true,
+	'menuitemradio': true,
+	'option': true,
+	'radio': true,
+	'slider': true,
+	'spinbutton': true,
+	'switch': true,
+	'tab:': true,
+	'textbox': true,
+	'treeitem': true
+};
+
+function getIsInteractiveChildClicked(e) {
+	const composedPath = e.composedPath();
+	const clickedElem = e.target;
+	for (let i = 0; i < composedPath.length; i++) {
+		const elem = composedPath[i];
+		if (elem === clickedElem) {
```

Comment:
Trying to wrap my mind around this check. I _think_ `e.target` will be the element that was clicked in the DOM scope in which the event handler is wired up.

So if the user clicks a `span` inside of a native `button` in the list-item's slot, `e.target` and `composedPath()[0]` would both be the `span`, and it would return `false` here ya?

Or, if the user clicks the `button` inside of `d2l-button` within the list-item's slot, the `e.target` would be `d2l-button` and the `composedPath()[0]` would be `button`, `slot`, or whatever element inside the slot they clicked. On one of the subsequent iterations of the loop we would return `false`, ya?

But in those cases we'd want to return true from this method.

Reply:
I guess the `e.target` is the link itself? So this is just bailing when we reach it.

Reply:
Trying to remember if there was a reason I did it this way but I believe you're right; in the link-item case we'd want it to go up to the link in the composed path and then return false.

The d2l-button case works because since it starts at the `button` it keeps going and since `button` is interactive it returns true.

Reply:
I was also thinking of the case where the consumer is extending `ListItemLinkMixin` themselves, and all the content is inside the list-item's shadowDOM instead of the lightDOM slot. Consumers have done that when they wanted their HM component to _be a_ list-item. Event retargeting would be my concern with the different scenarios, but I _think_ this solution would work fine for that case too.

Looks like [this demo](https://live.d2l.dev/BrightspaceUI/core/components/list/demo/list-nested.html#:~:text=d2l%2Dlist%3E-,Nested%20Custom%20Items,-3%20items) uses a custom item that could be tweaked if you need to check that.

Reply:
I tweaked that demo and all seemed to work fine